Player/GrenadeLauncher.gd

```diff
@@ -72,78 +71,73 @@ func update_aim() -> void:
 	var motion_down := to_target.y - peak_height
 	var time_going_down := sqrt(-2.0 * motion_down / gravity)
 	
-	var time_to_land := time_going_up + time_going_down
+	_time_to_land = time_going_up + time_going_down
 
 	var target_position_xz_plane := Vector3(to_target.x, 0.0, to_target.z)
 	var start_position_xz_plane := Vector3(_launch_point.position.x, 0.0, _launch_point.position.z)
 
-	var forward_velocity := (target_position_xz_plane - start_position_xz_plane) / time_to_land
+	var forward_velocity := (target_position_xz_plane - start_position_xz_plane) / _time_to_land
 	var velocity_up := sqrt(2.0 * gravity * motion_up)
 	
 	# Caching the found initial_velocity vector so we can use it on the throw() function
 	_throw_velocity = Vector3.UP * velocity_up + forward_velocity
 
-	# Redraw the grenade's motion preview
-#	_grenade_path.curve.clear_points()
-	_trail_points.clear()
+
+func _draw_throw_path() -> void:
 	const TIME_STEP := 0.05
-	var time_current := 0.0
-	var end_time := time_to_land + 0.5
-	while time_current < end_time:
-		var point := _throw_velocity * time_current + Vector3.DOWN * gravity * 0.5 * time_current * time_current
-		_trail_points.append(point)
-		time_current += TIME_STEP
-	_generate_mesh()
+	const TRAIL_WIDTH := 0.25
 
+	var forward_direction = Vector3(_throw_velocity.x, 0.0, _throw_velocity.z).normalized()
+	var left_direction := Vector3.UP.cross(forward_direction)
+	var offset_left = left_direction * TRAIL_WIDTH / 2.0
+	var offset_right = -left_direction * TRAIL_WIDTH / 2.0
 
-func _generate_mesh() -> void:
 	var st := SurfaceTool.new()
 	st.begin(Mesh.PRIMITIVE_TRIANGLES)
-	
-	# We'll use trail_direction to get the left and right points difference from
-	# the _trail_points 
-	var trail_direction = (_trail_points[-1] - _trail_points[0]).normalized()
-	trail_direction.y = 0.0
-	var cross_dir := Vector3.UP.cross(trail_direction)
-	var mesh_left_diff = cross_dir * TRAIL_SIZE/2.0
-	var mesh_right_diff = -cross_dir * TRAIL_SIZE/2.0
-
-	for i in range(_trail_points.size()-1):
-		# We'll create 2 triangles on each iteration, representing the quad of
-		# one section of the path (between two points of _trail_points)
-		
-		# _trail_points represents the points in the "center" of the path, so
-		# we'll derive the two points in the left and the two points in the right
-		var trail_point_left_1 = _trail_points[i] + mesh_left_diff
-		var trail_point_right_1 = _trail_points[i] + mesh_right_diff
-		var trail_point_left_2 = _trail_points[i+1] + mesh_left_diff
-		var trail_point_right_2 = _trail_points[i+1] + mesh_right_diff
+
+	var end_time := _time_to_land + 0.5
+	var point_previous = Vector3.ZERO
+	var time_current := 0.0
+	var uv_progress_factor := TIME_STEP / end_time
+	# We'll create 2 triangles on each iteration, representing the quad of one
+	# section of the path
+	while time_current < end_time:
```

Comment:
Instead of having two loops, calculating points in an array once, then looping over the array to draw, we can loop just once.